tests/test_connector.py

```diff
@@ -1,100 +1,145 @@
-        mocker.patch("phantom.base_connector.BaseConnector.add_action_result", return_value=ActionResult(dict(param)))
+        mock_result_adder.return_value = ActionResult(dict(param))
         mock_py42_client.users.get_current.side_effect = Py42UnauthorizedError(mock.Mock(status=401))
         set_status = mocker.patch("phantom.action_result.ActionResult.set_status")
 
         connector._handle_test_connectivity(param)
 
         set_status.assert_called_with(phantom.APP_ERROR, mocker.ANY)
 
-    def test_handle_action_calls_handle_connectivity(self, mocker, mock_py42_client, test_connectivity_connector):
+    def test_handle_action_calls_handle_connectivity(
+        self, mocker, test_connectivity_connector, mock_result_adder
+    ):
         def fake_set_status(self, status, message=None):
             return status == phantom.APP_SUCCESS
 
-        mocker.patch("py42.sdk.from_local_account", return_value=mock_py42_client)
         mocker.patch("phantom.action_result.ActionResult.set_status", new=fake_set_status)
         param = {}
-        mocker.patch("phantom.base_connector.BaseConnector.add_action_result", return_value=ActionResult(dict(param)))
+        mock_result_adder.return_value = ActionResult(dict(param))
 
         ret = test_connectivity_connector.handle_action(param)
 
         assert ret == phantom.APP_SUCCESS
 
-    def test_initialize_reads_credentials_from_config(self, mocker,
-                                                      mock_py42_client):
-        connector = code42_connector.Code42Connector()
-        config = {"cloud_instance": "http://cloud.code42.com", "username": "user.name", "password": "password123!"}
+    def test_initialize_reads_credentials_from_config(self, mocker):
+        connector = Code42Connector()
+        config = {
+            "cloud_instance": "http://cloud.code42.com",
+            "username": "user.name",
+            "password": "password123!"
+        }
         mocker.patch("phantom.base_connector.BaseConnector.get_config", return_value=config)
 
         connector.initialize()
 
         assert connector._cloud_instance == config["cloud_instance"]
         assert connector._username == config["username"]
         assert connector._password == config["password"]
+
+    def test_handle_action_when_add_departing_employee_calls_add_with_expected_args(
+        self, mocker, mock_py42_client, mock_result_adder
+    ):
+        param = {"username": "test@example.com", "departure_date": "2030-01-01"}
+        result = ActionResult(dict(param))
+        result.update_summary = mocker.MagicMock()
+        mock_result_adder.return_value = result
+        http_response = mocker.MagicMock(spec=Response)
+        http_response.text = json.dumps({"users": [{"userUid": "TEST_USER_UID"}]})
+        mock_py42_client.users.get_by_username.return_value = Py42Response(http_response)
+        connector = create_fake_connector(Code42Connector.ADD_DEPARTING_EMPLOYEE_ACTION_ID)
```

Comment:
I generally tend to avoid referencing constants that are in the project when writing tests. The main reason for this is because readers gain a better understanding of how the system is supposed to work when the input is not hidden behind a var that is defined in a separate project. For example, when I read this PR, I do not know what the id for this action actually is, but I would if it just appeared as a raw string here.

Reply:
I usually agree, and can put them like that if you want... My concern is that there are just so many raw strings here, it is maddening to know what it what.

Reply:
They have been swapped out 👍🏻